pseudocode-defns/pseudocode_defns.tex

```diff
@@ -0,0 +1,525 @@
+\end{definition}
+
+\begin{definition}[\texttt{get\_output\_domain}]
+    The function \texttt{get\_output\_domain(function)} returns the output domain of values returned by function \texttt{function}.
+\end{definition}
+
+\begin{definition}[\texttt{get\_output\_metric}]
+    The function \texttt{get\_output\_metric(some\_relation)} returns the output metric used by the relation \texttt{some\_relation}.
+\end{definition}
+
+\begin{definition}[\texttt{has\_trait}]
+    The function \texttt{has\_trait(T,(trait1,trait2,...))} is a function that returns $\True$ $\iffText$ the type \texttt{T} implements \texttt{trait1}, \texttt{trait2}, etc.
+\end{definition}
+
+\begin{definition}[\texttt{inf\_cast}]
+    This function is only defined for casting to types \texttt{TO} that have trait \texttt{InfCast(TI)}. The function \texttt{inf\_cast(val:TI, TO)} casts \texttt{val} to a value of type \texttt{TO} and returns that value. Specifically, \texttt{val} will be casted to the value of type \texttt{TO} that is closest to \texttt{val} and at least as large as \texttt{val}. If \texttt{inf\_cast} is not able to cast \texttt{val} to a value of type \texttt{TO} at least as large as \texttt{val}, then an error is returned instead.
+    
+    \textbf{Property:} \texttt{inf\_cast}ed distances are never less than input distances.
+\end{definition}
+
+\begin{definition}[\texttt{is\_instance}]
+    The function \texttt{is\_instance(var,T)} returns $\True$ if and only if the variable \texttt{var} is of type \texttt{T}.
+\end{definition}
+
+\begin{definition}[\texttt{len}]
+    The function \texttt{len(vector\_name)} returns the number of elements in \texttt{vector} \texttt{vector\_name}. Output is of type \texttt{usize}, so the return value $v$ on 32-bit machines is $v\in \{0,1,2,\ldots,2^{32} - 1\}$; likewise, the return value on 64-bit machines is $v\in \{0,1,2,\ldots,2^{64} - 1\}$. \inRust{std::vec::Vec::len}{https://doc.rust-lang.org/std/vec/struct.Vec.html\#method.len}
+\end{definition}
+
+\emph{Note: we do not call it \texttt{length} to avoid notational clashes with, for example, the Bounded Sum code.}
+
+\begin{definition}[\texttt{max}]
+    The function \texttt{max(var1, var2)} compares \texttt{var1} and \texttt{var2}, and returns the greater of the two values. When \texttt{var1} and \texttt{var2} are equivalent, it returns \texttt{var2}. \inRust{std::cmp::max}{https://doc.rust-lang.org/std/cmp/fn.max.html}
+\end{definition}
+
+\begin{definition}[\texttt{min}]
+    The function \texttt{min(var1:T, var2:T)} compares \texttt{var1} and \texttt{var2}, and returns the lesser of the two values. When \texttt{var1} and \texttt{var2} are equivalent, it returns \texttt{var1}. \inRust{std::cmp::min}{https://doc.rust-lang.org/std/cmp/fn.min.html}
+\end{definition}
+

```

Comment:
max and min should specify the types they operate on, specifically what traits are required.  Is it Ord or PartialOrd?  If PartialOrd, how are they defined on incomparable elements?

Reply:
They require TotalOrd -- added now.

Reply:
Resolving this comment because it appears to be fixed now.